service_configuration_lib/spark_config.py

```diff
@@ -980,6 +991,50 @@ def _convert_user_spark_opts_value_to_str(user_spark_opts: Mapping[str, Any]) ->
     return output
 
 
+def compute_approx_hourly_cost_dollars(spark_conf, paasta_pool):
+    per_executor_cores = int(spark_conf.get("spark.executor.cores", DEFAULT_EXECUTOR_CORES))
+    max_cores = per_executor_cores * (int(spark_conf.get("spark.executor.instances", DEFAULT_EXECUTOR_INSTANCES)))
+    min_cores = max_cores
+    if "spark.dynamicAllocation.enabled" in spark_conf and spark_conf["spark.dynamicAllocation.enabled"] == "true":
+        max_cores = per_executor_cores * (int(
+            spark_conf.get("spark.dynamicAllocation.maxExecutors", max_cores)
+        ))
+        min_cores = per_executor_cores * (int(
+            spark_conf.get("spark.dynamicAllocation.minExecutors", min_cores)
+        ))
+    if paasta_pool == "batch":
```

Comment:
We need to generalize this logic. 
Now we can have more pools beyond `batch` and `stable_batch`.

a) Create a global dict: Poo: Cost; Default -1 implying unknown or not calculated yet.
b) Read the pool list from puppet deployed config path